velox/expression/CastExpr.cpp

```diff
@@ -248,6 +248,46 @@ VectorPtr CastExpr::castToDate(
   }
 }
 
+VectorPtr CastExpr::castIntToTimestamp(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType) {
+  VectorPtr castResult;
+  context.ensureWritable(rows, TIMESTAMP(), castResult);
+  (*castResult).clearNulls(rows);
+  auto* resultFlatVector = castResult->as<FlatVector<Timestamp>>();
+  const auto& queryConfig = context.execCtx()->queryCtx()->queryConfig();
+  const auto sessionTzName = queryConfig.sessionTimezone();
+  const auto* timeZone =
+      (queryConfig.adjustTimestampToTimezone() && !sessionTzName.empty())
+      ? date::locate_zone(sessionTzName)
+      : nullptr;
```

Comment:
It seems we can combine above code into castIntegerToTimestamp, and dispatch in applyPeeled once. Do you think that would make the implementation more concise?